pkg/providers/instancetype/types.go

```diff
@@ -564,6 +564,18 @@ func pods(ctx context.Context, info ec2types.InstanceTypeInfo, amiFamily amifami
 	if lo.FromPtr(podsPerCore) > 0 && amiFamily.FeatureFlags().PodsPerCoreEnabled {
 		count = lo.Min([]int64{int64(lo.FromPtr(podsPerCore)) * int64(lo.FromPtr(info.VCpuInfo.DefaultVCpus)), count})
 	}
+
+	// sig-scalability and aws recommend capping pods, see:
+	// - https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md
+	// - https://github.com/awslabs/amazon-eks-ami/blob/main/templates/al2/runtime/max-pods-calculator.sh
+	// - https://github.com/aws/karpenter-provider-aws/issues/8286
+	// TODO: make this opt-in or opt-out
+	if int64(lo.FromPtr(info.VCpuInfo.DefaultVCpus)) > 30 {
```

Comment:
that's not what our script does though. 110 is max ceiling for smaller nodes and 250 is max ceiling for larger nodes.

but the max ips per node is still a limit

```
❯ ./max-pods-calculator.sh \
    --instance-type m6i.2xlarge \
    --cni-version 1.19.6 \
    --cni-custom-networking-enabled
44
```

```
 ./max-pods-calculator.sh \
    --instance-type g4dn.xlarge \
    --cni-version 1.19.6 \
    --cni-custom-networking-enabled
20
```